features/docs/writing_support_code/world_namespaces.feature

```diff
@@ -0,0 +1,186 @@
+Feature: World Namespaces
+
+  In order to better organise the code supporting my tests
+  As a cucumber user
+  I want to access the method added to a World using an explicit namespace
+
+  Scenario: A single module is imported with namespace
+    Given a file named "features/support/helpers.rb" with:
+      """
+      module Helpers
+        def helper_method
+          42
+        end
+      end
+
+      World(my_namespace: Helpers)
+      """
+    And a file named "features/step_definitions/step.rb" with:
+      """
+      Then /^the helper method is called$/ do
+        expect(my_namespace.helper_method).to eql(42)
+      end
+      """
+    And a file named "features/f.feature" with:
+      """
+      Feature: Calling a method
+        Scenario: I call a method from a namespace
+          Then the helper method is called
+      """
+    When I run `cucumber features/f.feature`
+    Then it should pass
+
+  Scenario: Multiple modules are imported with a namespace
+    Given a file named "features/support/helpers.rb" with:
+      """
+      module ModuleOne
+        def forty_one
+          41
+        end
+      end
+
+      module ModuleTwo
+        def forty_two
+          42
+        end
+      end
+
+      World(module_one: ModuleOne, module_two: ModuleTwo)
+      """
+    And a file named "features/step_definitions/step.rb" with:
+      """
+      Then /^the helper method is called$/ do
+        expect(module_one.forty_one).to eql(41)
+        expect(module_two.forty_two).to eql(42)
+      end
+      """
+    And a file named "features/f.feature" with:
+      """
+      Feature: Calling a method
+        Scenario: I call a method from two namespaces
+          Then the helper method is called
+      """
+    When I run `cucumber features/f.feature`
+    Then it should pass
+
+  Scenario: Assigning different modules to the same namespace
```

Comment:
While it's useful to know that this works, it does seem like an edge case. Is there a way to push this down to a unit test?